lxd/auth/authorization.go

```diff
@@ -0,0 +1,73 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/request"
+	"github.com/canonical/lxd/shared"
+)
+
+type Authorizer interface {
+	AddProject(id int64, name string) error
+	DeleteProject(id int64) error
+	RenameProject(id int64, newName string) error
+
+	StopStatusCheck()
+
+	UserAccess(username string) (*UserAccess, error)
+	UserIsAdmin(r *http.Request) bool
+	UserHasPermission(r *http.Request, projectName string, permission string) bool
+}
+
+// UserAccess struct for permission checks.
+type UserAccess struct {
+	Admin    bool
+	Projects map[string][]string
+}
+
+type CommonAuthorizer struct{}
+
+func (a *CommonAuthorizer) AddProject(id int64, name string) error {
+	return nil
```

Comment:
Does nil mean the user can perform the action? If so can you add a comment here explaining that.

Reply:
I've added a comment explaining what this function is supposed to do.